I got another test failure during our weekly python 3.9 update on Arch Linux:
```
======================================================================
FAIL: test_pidfd_send_signal (test.test_signal.PidfdSignalTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/python-git/src/cpython/Lib/test/test_signal.py", line 1287, in test_pidfd_send_signal
    self.assertEqual(cm.exception.errno, errno.EBADF)
AssertionError: 1 != 9

----------------------------------------------------------------------
```
Could that test be skipped in case of EPERM from the system call `pidfd_send_signal`, too?

--- a/Lib/test/test_posix.py
+++ b/Lib/test/test_posix.py
@@ -1476,6 +1476,8 @@ class PosixTester(unittest.TestCase):
             os.pidfd_open(-1)
         if cm.exception.errno == errno.ENOSYS:
             self.skipTest("system does not support pidfd_open")
+        if isinstance(cm.exception, PermissionError):
+            self.skipTest(f"pidfd_open syscall blocked: {cm.exception!r}")
         self.assertEqual(cm.exception.errno, errno.EINVAL)
         os.close(os.pidfd_open(os.getpid(), 0))
 

--- a/Misc/NEWS.d/next/Tests/2019-11-20-15-42-06.bpo-38692.aqAvyF.rst
+++ b/Misc/NEWS.d/next/Tests/2019-11-20-15-42-06.bpo-38692.aqAvyF.rst
@@ -1,0 +1,3 @@
+Skip the test_posix.test_pidfd_open() test if ``os.pidfd_open()`` fails with a
+:exc:`PermissionError`. This situation can happen in a Linux sandbox using a
+syscall whitelist which doesn't allow the ``pidfd_open()`` syscall yet.

